src/main/java/io/github/fablabsmc/fablabs/api/fluidvolume/v1/volume/api/FluidVolume.java

```diff
@@ -0,0 +1,224 @@
+package io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.api;
+
+import java.util.AbstractCollection;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Objects;
+
+import com.google.common.collect.Iterators;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.math.Fraction;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.properties.FluidPropertyManager;
+
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.Fluids;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+/**
+ * a mutable representation of a quantity and it's data of a fluid.
+ * this class can and should be extended for custom logic when dealing with volume merging.
+ *
+ * @see #merge(FluidVolume)
+ * @see #drain(FluidVolume)
+ * @see #draw(Fraction)
+ * @see SimpleFixedSizedFluidVolume
+ */
+public class FluidVolume extends AbstractCollection<FluidContainer> implements FluidContainer, SingleFluidContainer {
```

Comment:
I find it slightly odd the structure of how FluidVolume is a FluidContainer...but I suppose it somewhat makes sense. I just more expected a paradigm closer to the vanilla Inventory interface for items. I'll wait until I try using this api until I decide whether or not I like it

Reply:
FluidHandler ⚪

Reply:
actually yeah, FluidHandler seems like a better name than FluidContainer